pocketutils/misc/klgists.py

```diff
@@ -0,0 +1,829 @@
+	:param path: Read the file at this local path
+	:return: A dict mapping keys to values, both with surrounding whitespace stripped
+	"""
+	lines = read_lines_file(path, ignore_comments=False)
+	dct = {}
+	for i, line in enumerate(lines):
+		if line.startswith('#'): continue
+		if line.count('=') != 1:
+			raise ParsingError("Bad line {} in {}".format(i+1, path))
+		parts = line.split('=')
+		dct[parts[0].strip()] = parts[1].strip()
+	return dct
+
+
+class Comparable:
+	"""A class that's comparable. Just implement __lt__. Credit ot Alex Martelli on https://stackoverflow.com/questions/1061283/lt-instead-of-cmp"""
+
+	def __eq__(self, other):
+		return not self < other and not other < self
+
+	def __ne__(self, other):
+		return self < other or other < self
+
+	def __gt__(self, other):
+		return other < self
+
+	def __ge__(self, other):
+		return not self < other
+
+	def __le__(self, other):
+		return not other < self
+
+
+def json_serial(obj):
+	"""JSON serializer for objects not serializable by default json code.
+	From jgbarah at https://stackoverflow.com/questions/11875770/how-to-overcome-datetime-datetime-not-json-serializable
+	"""
+	if isinstance(obj, (datetime, date)):
+		return obj.isoformat()
+	try:
+		import peewee
+		if isinstance(obj, peewee.Field):
+			return type(obj).__name__
+	except ImportError: pass
+	raise TypeError("Type %s not serializable" % type(obj))
+
+def pretty_dict(dct: dict) -> str:
+	"""Returns a pretty-printed dict, complete with indentation. Will fail on non-JSON-serializable datatypes."""
+	return json.dumps(dct, default=json_serial, sort_keys=True, indent=4)
+
+def pp_dict(dct: dict) -> None:
+	"""Pretty-prints a dict to stdout."""
+	print(pretty_dict(dct))
+
+def pp_size(obj: object) -> None:
+	"""Prints to stdout a human-readable string of the memory usage of arbitrary Python objects. Ex: 8M for 8 megabytes."""
+	print(hsize(sys.getsizeof(obj)))
+
+def sanitize_str(value: str) -> str:
+	"""Removes Unicode control (Cc) characters EXCEPT for tabs (\t), newlines (\n only), line separators (U+2028) and paragraph separators (U+2029)."""
+	return "".join(ch for ch in value if unicodedata.category(ch) != 'Cc' and ch not in {'\t', '\n', '\u2028', '\u2029'})
+
+def escape_for_properties(value: Any) -> str:
+	return sanitize_str(str(value).replace('\n', '\u2028'))
+
+def escape_for_tsv(value: Any) -> str:
+	return sanitize_str(str(value).replace('\n', '\u2028').replace('\t', ' '))
+	
+class Timeout:
+	def __init__(self, seconds: int = 10, error_message='Timeout'):
+		self.seconds = seconds
+		self.error_message = error_message
+	def handle_timeout(self, signum, frame):
+		raise TimeoutError(self.error_message)
+	def __enter__(self):
+		signal.signal(signal.SIGALRM, self.handle_timeout)
+		signal.alarm(self.seconds)
+	def __exit__(self, type, value, traceback):
+		signal.alarm(0)
+
+
+class OverwriteChoice(Enum):
+	FAIL = 1
+	WARN = 2
+	IGNORE = 3
+	OVERWRITE = 4
+
+def fix_path(path: str) -> str:
```

Comment:
[PathTools `sanitize_path`](https://github.com/dmyersturnbull/pocketutils/blob/main/pocketutils/tools/path_tools.py#L101). ([`sanitize_nodes`](https://github.com/dmyersturnbull/pocketutils/blob/main/pocketutils/tools/path_tools.py#L143) is better.)